std/algorithm/searching.d

```diff
@@ -1361,7 +1361,19 @@ if (isInputRange!InputRange &&
 
     alias EType  = ElementType!R;
 
-    static if (isNarrowString!R)
+    // If the haystack is a SortedRange we can use binary search to find the needle.
+    // Works only for the default find predicate and any SortedRange predicate.
+    // 8829 enhancement
+    import std.range: SortedRange;
+    static if(is(typeof(haystack) : SortedRange!TT, TT) && isDefaultPred)
```

Comment:
You already have the type of haystack: `static if (is(InputRange : SortedRange!TT, TT) && isDefaultPred)`